File: tests/csharp/test_standard.cs
/**
 * Test entry point for standard message tests (C#).
 */

using System;
using StructFrame.SerializationTest;
using StructFrameTests;

class TestStandard
{
    private const string TEST_NAME = "StandardMessages";
    private const string PROFILES = "standard, sensor, ipc, bulk, network";

    public static int Main(string[] args)
    {
        // Verify enum ToString() before running the main test suite
        if (!StandardMessages.CheckEnumToString())
        {
            Console.Error.WriteLine("[FAIL] C# enum ToString() check failed");
            return 1;
        }

        // Verify discriminator=none oneof round-trip
        if (!StandardMessages.CheckDiscriminatorNone())
        {
            Console.Error.WriteLine("[FAIL] C# discriminator=none oneof check failed");
            return 1;
        }

        // Verify multiple oneof fields in one message
        if (!StandardMessages.CheckMultiOneof())
        {
            Console.Error.WriteLine("[FAIL] C# multi-oneof check failed");
            return 1;
        }

        return TestHarness.Run(
            args,
            StandardMessages.MESSAGE_COUNT,
            StandardMessages.GetMessage,
            StandardMessages.CheckMessage,
            (msgId) => MessageDefinitions.GetMessageInfo(msgId),
            TEST_NAME,
            PROFILES
        );
    }
}

File: tests/ts/test_standard.ts
/**
 * Test entry point for standard message tests (TypeScript).
 */

import { MESSAGE_COUNT, getMessage, checkMessage, checkEnumToString, checkDiscriminatorNone, checkMultiOneof } from './include/standard_messages';
import { getMessageInfo } from '../generated/ts/serialization-test.structframe';
import { run } from './include/test_harness';

const TEST_NAME = 'StandardMessages';
const PROFILES = 'standard, sensor, ipc, bulk, network';

// Verify enum-to-string (reverse mapping) before running the main test suite
if (!checkEnumToString()) {
  console.error('[FAIL] TypeScript enum-to-string reverse mapping check failed');
  process.exit(1);
}

// Verify discriminator=none oneof round-trip
if (!checkDiscriminatorNone()) {
  console.error('[FAIL] TypeScript discriminator=none oneof check failed');
  process.exit(1);
}

// Verify multiple oneof fields in one message
if (!checkMultiOneof()) {
  console.error('[FAIL] TypeScript multi-oneof check failed');
  process.exit(1);
}

process.exit(run(MESSAGE_COUNT, getMessage, checkMessage, getMessageInfo, TEST_NAME, PROFILES));
